site/src/layouts/DetailComponent/DetailComponent.tsx

```diff
@@ -0,0 +1,66 @@
+import React, { FC, useCallback, useMemo } from "react";
+import { useRouter } from "next/navigation";
+import {
+  TableOfContents,
+  Sidebar,
+} from "@jpmorganchase/mosaic-site-components";
+import { useRoute, useStore, SiteState } from "@jpmorganchase/mosaic-store";
+import { TabPanel, Tabs } from "@salt-ds/lab";
+import { LayoutProps } from "../types/index";
+import { DetailBase } from "../DetailBase";
+
+const tabs = [
+  { id: 0, name: "examples", label: "Examples" },
+  { id: 1, name: "usage", label: "How to use" },
+  { id: 2, name: "accessibility", label: "Accessibility" },
+];
+
+type CustomSiteState = SiteState & { data?: Record<string, any> };
+
+export const DetailComponent: FC<LayoutProps> = ({ children }) => {
+  const { push } = useRouter();
+  const { route } = useRoute();
+
+  const newRoute = route?.substring(0, route.lastIndexOf("/") + 1);
+
+  const useData = useStore((state: CustomSiteState) => state.data);
+
+  const { description } = useData || {};
+
+  const updateRouteWhenTabChanges = useCallback((index: number) => {
+    const currentTab = tabs.find(({ id }) => index === id);
+
+    currentTab
+      ? push(`${newRoute}${currentTab.name}`)
+      : push(`${newRoute}${tabs[0].name}`);
+  }, []);
+
+  const updateTabWhenRouteChanges = useMemo(() => {
+    const currentTab = tabs.find(({ name }) => route?.includes(name));
+
+    const defaultTabAndRoute = () => {
+      push(`${newRoute}${tabs[0].name}`);
+      return 0;
+    };
+
+    return currentTab ? currentTab.id : defaultTabAndRoute();
+  }, [route]);
+
+  const SecondarySidebar = <TableOfContents />; // TODO: replace with custom component pages sidebar
+
```

Comment:
I would have expected the code to be something like:

```tsx
const currentTab = tabs.find(({ name }) => route?.includes(name));
const currentTabIndex = currentTab?.id ?? 0
```

Reply:
This would be fine but we also need to update the url when setting the tab to 0 hence the `defaultTabAndRoute` function

Reply:
I'm not sure if we need to change the URL. I think it's better to keep the root and the explicit separate

Reply:
Okay makes sense, I'll update this to only default the tab to 0 but maintain the url

Reply:
@joshwooding as discussed I've moved this redirect into a `useEffect`